src/lib/middleware.ts

```diff
@@ -31,6 +41,48 @@ export function withCookieRedirect<T>(callback: GetServerSideProps<T>) {
   };
 }
 
+/**
+ * Decorator function to add beacons specific information to the `getServerSideProps` context.
+ *
+ * @param context {GetServerSidePropsContext}   The NextJS application context
+ * @returns       {Promise<BeaconsContext>}     A promise resolving to the decorated context containing application specific data
+ */
+export async function decorateGetServerSidePropsContext(
```

Comment:
This makes life a lot easier doing calculations in each page like:

- What is the current cache entry
- Has the user accepted the cookie policy
- What is the data parsed from the form

As the objects live as top level keys on the `getServerSideProps` Context object. It worked nicely for updating the `handlePageRequest`, only thing I would note is that it might be worth splitting this file up into more meaningful names if it starts to grow